reactor-core/src/main/java/reactor/core/publisher/FluxReplay.java

```diff
@@ -1380,8 +1403,10 @@ public boolean isDisposed() {
 				AtomicLongFieldUpdater.newUpdater(ReplayInner.class, "requested");
 
 
-		ReplayInner(CoreSubscriber<? super T> actual) {
+		ReplayInner(CoreSubscriber<? super T> actual, ReplaySubscriber<T> parent) {
 			this.actual = actual;
+			this.parent = parent;
+			this.registeredBeforeConnection = ReplaySubscriber.CONNECTED.get(parent) == 0;
```

Comment:
nit: since `ReplayInner ` isn't reused, WDYT about accepting `boolean registeredBeforeConnection` and doing `ReplaySubscriber.CONNECTED.get(parent) == 0` at the instantiation time?
Although the class is internal, `ReplaySubscriber.CONNECTED` usage here looks a bit unexpected

Reply:
improved the situation to avoid that, at the expense of additional state - which proves useful in obtaining a correct implementation actually 👍